### PR TITLE
create book and bookRepository

### PR DESCRIPTION
**Pull request by Mariana-VV**

### CODE REVIEW

File: src/main/java/org/bookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
```
Comment (koval-kn) [inline src/main/java/org/bookstore/model/Book.java]: dont use data along with entity annotation as it cause bugs
should be replaced with getter/setter

File: src/main/java/org/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (koval-kn) [inline src/main/java/org/bookstore/model/Book.java]: dont forget to add required constraints (not null unique etc)

File: src/main/java/org/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public Book(String title, String author, String isbn,
+                BigDecimal price, String description,
+                String coverImage) {
+        this.title = title;
+        this.author = author;
+        this.isbn = isbn;
+        this.price = price;
+        this.description = description;
+        this.coverImage = coverImage;
+    }
```
Comment (koval-kn) [inline src/main/java/org/bookstore/model/Book.java]: use `@NoArgsConstructor` and `@AllArgsConstructor` annotations instead
```suggestion
```

File: src/main/java/org/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        return null;
+    }
+
+    @Override
+    public List getAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("select b from Book b").getResultList();
+        }
+
```
Comment (koval-kn) [inline src/main/java/org/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1 +1,9 @@
 spring.application.name=BookStoreApp
+spring.datasource.url=jdbc:mysql://localhost:3306/bookstore?serverTimezone=UTC
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.username=root
+spring.datasource.password=aqwa2012
+
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
+
```
Comment (koval-kn) [inline src/main/resources/application.properties]: you should have exactly 1 blank line at the EOF
```suggestion
```

File: src/main/java/org/bookstore/service/BookService.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package org.bookstore.service;
+
+import java.util.List;
+import org.bookstore.model.Book;
+
+public interface BookService {
+    Book save(Book book);
+
+    List<Book> getAll();
```
Comment (koval-kn) [inline src/main/java/org/bookstore/service/BookService.java]: image

```suggestion
    Book save(Book book);

    List<Book> findAll();
```

File: src/main/java/org/bookstore/repository/BookRepository.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package org.bookstore.repository;
+
+import java.util.List;
+import org.bookstore.model.Book;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    List<Book> getAll();
```
Comment (koval-kn) [inline src/main/java/org/bookstore/repository/BookRepository.java]: image
```suggestion
    Book save(Book book);

    List<Book> findAll();
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]
Good job!

**Pull request merged**